### PR TITLE
MRG+1: Unit-gain LCMV beamformer

### LINKED ISSUE #6183: MAINT: Beamformer comparisons
As discussed with @larsoner, we will add a test for LCMV and DICS to test against FieldTrip (i.e., against saved FieldTrip output).

### PR DESCRIPTION
**Pull request by britta-wstnr**
This fixes the pseudo-inversion in the filter computation.
In the old code, the ``pinv`` had an arbitray ``cond=0.1`` set. We already stumbled across this in the past several times (the code is from before we started the overhaul of the module). So far we kept it, since we had no good motivation to change it.
Now, I ran our beamformer against FieldTrip, and if I get rid of this threshold ``cond=0.1``, the variance explained between the toolboxes for the beamformer output **jumps from 0% to 100%** and **the output gets much more plausible than before**.

After investigating this, I would consider the old code a bug.

Of course, we'd have to change the value ranges for the unit tests now ...

Thoughts? @larsoner @agramfort @wmvanvliet 

Edit: I added the FieldTrip comparisons here as well - closes #6183

### CODE REVIEW

**Comment by wmvanvliet** [comment]
+100 from me on this!

File: mne/beamformer/tests/test_external.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+fname_fwd_vol = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc-meg-vol-7-fwd.fif')
+fname_event = op.join(data_path, 'MEG', 'sample',
+                      'sample_audvis_trunc_raw-eve.fif')
+fname_label = op.join(data_path, 'MEG', 'sample', 'labels', 'Aud-lh.label')
+
+reject = dict(grad=4000e-13, mag=4e-12)
+
+# where FieldTrip data should be saved to / imported from
+ft_data_path = ' '
```
Comment (britta-wstnr) [inline mne/beamformer/tests/test_external.py]: @larsoner where should the data live?

File: mne/beamformer/tests/test_external.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        stc_ft = mne.read_source_estimate(ft_fname)
+
+        # calculate the Pearson correlation between the source solutions:
+        pearson = np.corrcoef(np.concatenate(stc_mne.data),
+                              np.concatenate(stc_ft.data))
+
+        assert pearson[0, 1] >= 0.99
+
+
+test_lcmv_fieldtrip()
```
Comment (britta-wstnr) [inline mne/beamformer/tests/test_external.py]: Is it right that this needs to be changed in ``run_tests_if_main()`` ?  Do I have to add the test somewhere for it to work?

File: mne/beamformer/tests/test_external.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        savemat(cov_savepath, sample_cov)
+        # evoked data:
+        ev_savepath = op.join(ft_data_path, 'sample_evoked')
+        data_ev = {'sample_evoked': evoked.data}
+        savemat(ev_savepath, data_ev)
+
+    return evoked, data_cov, fwd
+
+
+def test_lcmv_fieldtrip():
```
Comment (britta-wstnr) [inline mne/beamformer/tests/test_external.py]: Does this need a decorator ``@testing.requires_testing_data`` ?

File: mne/beamformer/tests/test_external.py (hunk 2 above)
Comment (larsoner) [reply mne/beamformer/tests/test_external.py]: Don't call the test manually, let `pytest mne/...` do it. `run_tests_if_main()` in theory should do this if you run the file as a script but in practice this mode is currently broken for many tests. So feel free to add `run_tests_if_main` to the bottom for future compat

File: mne/beamformer/tests/test_external.py (hunk 1 above)
Comment (larsoner) [reply mne/beamformer/tests/test_external.py]: I would make it `MNE-testing-data/fieldtrip/*`

File: mne/beamformer/tests/test_external.py (hunk 3 above)
Comment (larsoner) [reply mne/beamformer/tests/test_external.py]: Yes

**Comment by larsoner** [comment]
I'll push a commit to fix the tests unless you're working on it now

**Comment by britta-wstnr** [comment]
Thanks for the tests, @larsoner ! Just fixed the beamformer name in ``whats_new``.
Do you want to have a look, @wmvanvliet ?

**Comment by larsoner** [comment]
@britta-wstnr don't worry about changing the conditional for `dics` here regarding `weight_norm=None, inversion='matrix'` in tests here, I looked and I think it's already mostly doing that.

**Comment by britta-wstnr** [comment]
I also made sure now that the ill-defined combination ``weight_norm='unit-noise-gain', normalize_fwd=True, inversion='single'`` throws an error, prompting the user to drop the forward normalization.
That involved fixing some tests, but I only added ``normalize_fwd=True``, no assertion values were changed.
I also added a test for the error message.

**Comment by larsoner** [comment]
Good to go from your end?

The renaming of `normalize_fwd` should be a subsequent PR

**Comment by larsoner** [comment]
Ahh I saw the change at the project end, so have retitled accordingly

**Comment by britta-wstnr** [comment]
Yes, I think I am done here, let's hope we come back green!

**Comment by wmvanvliet** [comment]
MEG+1 from me.

**Comment by larsoner** [comment]
Thanks @britta-wstnr !

**Pull request merged**